Applications/Calculator/CalculatorWindow.gml

```diff
@@ -0,0 +1,272 @@
+            text: "6"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+
+        @GUI::Button {
+            name: "multiply_button"
+            text: "*"
+            fixed_width: 35
+            fixed_height: 28
+        }
+
+        @GUI::Button {
+            name: "mod_button"
+            text: "%"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+    }
+
+    @GUI::Widget {
+
+        layout: @GUI::HorizontalBoxLayout {}
+
+        @GUI::Button {
+            name: "mem_save_button"
+            text: "MS"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "red"
+        }
+
+        @GUI::Widget {
+            fixed_width: 5
+        }
+
+        @GUI::Button {
+            name: "1_button"
+            text: "1"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+
+        @GUI::Button {
+            name: "2_button"
+            text: "2"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+
+        @GUI::Button {
+            name: "3_button"
+            text: "3"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+
+        @GUI::Button {
+            name: "subtract_button"
+            text: "-"
+            fixed_width: 35
+            fixed_height: 28
+        }
+
+        @GUI::Button {
+            name: "inverse_button"
+            text: "1/x"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+    }
+
+    @GUI::Widget {
+
+        layout: @GUI::HorizontalBoxLayout {}
+
+        @GUI::Button {
+            name: "mem_add_button"
+            text: "M+"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "red"
+        }
+
+        @GUI::Widget {
+            fixed_width: 5
+        }
+
+        @GUI::Button {
+            name: "0_button"
+            text: "0"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+
+        @GUI::Button {
+            name: "sign_button"
+            text: "+/-"
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+
+        @GUI::Button {
+            name: "decimal_button"
+            text: "."
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "blue"
+        }
+
+        @GUI::Button {
+            name: "add_button"
+            text: "+"
+            fixed_width: 35
+            fixed_height: 28
+        }
+
+        @GUI::Button {
+            name: "equal_button"
+            text: "="
+            fixed_width: 35
+            fixed_height: 28
+            foreground_color: "red"
+        }
+    }
+}
```

Comment:
There is not a nonew line at the end of the file

Reply:
there should be one?

Reply:
Yes, it's good practice to always have one. In order to not have your prompt messed up if you `cat` this file. (all the other files in this patch have only this doesn't)

Reply:
Will keep that in mind. added